src/config/graph.rs

```diff
@@ -283,6 +283,48 @@ impl Graph {
             .map(|edge| edge.from.clone())
             .collect()
     }
+
+    /// From a given root node, get all paths from the root node to leaf nodes
+    /// where the leaf node must be a sink. This is useful for determining which
+    /// components are relevant in a Vector unit test.
+    ///
+    /// Caller must check for cycles before calling this function.
+    pub fn paths_to_sink_from(&self, root: &ComponentKey) -> HashSet<ComponentKey> {
```

Comment:
Yes, I'll add tests as this is a particularly critical piece in the new refactor. If the graph has cycles, this function can potentially loop forever. I debated calling `check_for_cycles` within `paths_to_sink_from` but left it up to the caller since you only need to check for cycles once per graph whereas you may call this function multiple times per graph.